ambry-store/src/test/java/com.github.ambry.store/StoreFindTokenTest.java

```diff
@@ -168,35 +165,54 @@ public void constructionErrorCasesTest() {
     } catch (IllegalArgumentException e) {
       // expected. Nothing to do.
     }
-
-    // no offset in IndexBased
-    try {
-      new StoreFindToken(key, null, sessionId, null);
-      fail("Construction of StoreFindToken should have failed");
-    } catch (IllegalArgumentException e) {
-      // expected. Nothing to do.
-    }
-
-    // no sessionId in IndexBased
-    try {
-      new StoreFindToken(key, offset, null, null);
-      fail("Construction of StoreFindToken should have failed");
-    } catch (IllegalArgumentException e) {
-      // expected. Nothing to do.
-    }
   }
 
   // helpers
   // general
 
   /**
-   * Compares two tokens to ensure their equality test passes and that their hash codes match.
+   * Compares two tokens to ensure their equality test passes and that their hash codes matches for recent
+   * versions
+   * @param reference the reference {@link StoreFindToken}
+   * @param toCheck the {@link StoreFindToken} to check
+   * @param version the StoreFindToken version to be tested
+   */
+  private void compareTokens(StoreFindToken reference, StoreFindToken toCheck, short version) {
+    switch (version) {
```

Comment:
Since serialization of storeToken was still happening in V1(before this patch), no tests were added to test V_2 for serialization and deserialization. In other words, ser and deser wasn't tested with incarnationId before this patch.